File: remote-hotrod-rest-compatibility/src/main/java/org/littlewings/infinispan/compatibility/Person.java
package org.littlewings.infinispan.compatibility;

import java.io.Serializable;
import java.util.Objects;

public class Person implements Serializable {
    private String firstName;
    private String lastName;

    public Person(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstName, lastName);
    }

    @Override
    public boolean equals(Object other) {
        if (other instanceof Person) {
            Person p = (Person) other;
            return Objects.equals(firstName, p.firstName) && Objects.equals(lastName, p.lastName);
        } else {
            return false;
        }
    }
}

File: embedded-remote-compatibility-mode/src/test/scala/org/littlewings/infinispan/compatibility/Person.scala
package org.littlewings.infinispan.compatibility

import java.util.Objects

class Person(val name: String, val age: Int) extends Serializable {
  override def equals(o: Any): Boolean =
  o match {
    case other: Person => Objects.equals(name, other.name) && Objects.equals(age, other.age)
    case _ => false
  }

  override def hashCode: Int = Objects.hash(name, Integer.valueOf(age))
}
